DeAceroDLabs/marketplace-frontend
# [DMP-429 | DMP-431] Login View & Redirect to Marketplace

## What is done

[[DMP-420] Login View and Redirect](https://deacero.atlassian.net/browse/DMP-420)

## How is done

1. Added a new Login component where the Login page will live.
2. Created a new React Context where the user data will live once this is logged in.
3. Created a ProtectedRoute component that verifies that the user is logged in before going to any page.

## How to test

1. Run the project via `npm start`
4. You will be sent to the Login Page.
5. Set any username in the username box (the password doesn't matter for now).
6. Once you click on login or press Enter, you will be redirected to the home page.
7. In the Home header, you will see your name (or the username you used in the login).
8. Refresh the page to lose the user info.
9. Try to navigate to `/` or `/search/something`, you should be redirected to `/login`
